consensus/types/src/beacon_state.rs

```diff
@@ -613,6 +613,27 @@ impl<T: EthSpec> BeaconState<T> {
         cache.get_all_beacon_committees()
     }
 
+    /// Returns the block root which decided the proposer shuffling for the epoch passed in parameter. This root
+    /// can be used to key this proposer shuffling.
+    ///
+    /// ## Notes
+    ///
+    /// The `block_root` covers the one-off scenario where the genesis block decides its own
+    /// shuffling. It should be set to the latest block applied to `self` or the genesis block root.
+    /// The `epoch` should be set to the epoch of the block root.
```

Comment:
```suggestion
    /// The `block_root` must be equal to the latest block applied to `self`.
```

Since we can provide an `epoch` greater than `self.current_epoch()` the `block_root` is used for scenarios other than just the genesis "one-off scenario".